recipes-bsp/u-boot/u-boot/0005-generate-random-ethaddr-by-hardcoding-invalid-addres.patch

```diff
@@ -0,0 +1,37 @@
+From 6affd0f9f0f19cdfb44d5483c7d73d14f2185b0a Mon Sep 17 00:00:00 2001
+From: Jakub Lecki <jakub.lecki@3mdeb.com>
+Date: Mon, 26 Jul 2021 10:37:28 +0200
+Subject: [PATCH] generate random ethaddr by hardcoding invalid address
+
+Generated address will be stored in environment variables
+
+Signed-off-by: Jakub Lecki <jakub.lecki@3mdeb.com>
+---
+ include/configs/sunxi-common.h | 1 +
+ net/eth-uclass.c               | 1 +
+ 2 files changed, 2 insertions(+)
+
+diff --git a/include/configs/sunxi-common.h b/include/configs/sunxi-common.h
+index 0ef289fd64..2263aba169 100644
+--- a/include/configs/sunxi-common.h
++++ b/include/configs/sunxi-common.h
+@@ -500,6 +500,7 @@ extern int soft_i2c_gpio_scl;
+ 	"uuid_gpt_system=" UUID_GPT_SYSTEM "\0" \
+ 	"partitions=" PARTS_DEFAULT "\0" \
+ 	BOOTCMD_SUNXI_COMPAT \
++	"ethaddr=00:00:00:00:00:00\0" \
+ 	BOOTENV
+ 
+ #else /* ifndef CONFIG_SPL_BUILD */
+diff --git a/net/eth-uclass.c b/net/eth-uclass.c
+index ed81cbd537..ba64446563 100644
+--- a/net/eth-uclass.c
++++ b/net/eth-uclass.c
+@@ -551,6 +551,7 @@ static int eth_post_probe(struct udevice *dev)
+ 		net_random_ethaddr(pdata->enetaddr);
+ 		printf("\nWarning: %s (eth%d) using random MAC address - %pM\n",
+ 		       dev->name, dev->seq, pdata->enetaddr);
++		eth_env_set_enetaddr_by_index("eth", dev->seq, pdata->enetaddr);
```

Comment:
@leckijakub Let's add some comments also here, maybe a small printf informing that ethaddr was saved in the environment variable

Reply:
@TomaszAIR comment along with a `printf` added